### PR TITLE
[TOMEE-4053] Dependency properties cleanup, step 1 and 2 on branch 9.x

### PR DESCRIPTION
**Pull request by sultan**
this branch has the properties re-ordered/regrouped by categories, with name changes, we can test this on ci build.

### CODE REVIEW

**Review by rzo1 — APPROVED** [review]

**Comment by sultan** [comment]
to be merged after 
* PR #957

**Comment by sultan** [comment]
rebased and ready

**Comment by sultan** [comment]
added a step 2 with a few name changes that passed OK on the previous version branch

File: pom.xml [hunk 1]
```diff
@@ -1341,7 +1327,7 @@
       <dependency>
         <groupId>org.apache.geronimo.mail</groupId>
         <artifactId>geronimo-mail_2.1_provider</artifactId>
-        <version>${geronimo-mail_2.1_provider.version}</version>
+        <version>${version.geronimo-javamail_1.6_mail}</version>
```
Comment (rzo1) [inline pom.xml:1330]: This version string doesn't look right? Might explain my confusion regarding one of the dropped properties.

File: pom.xml [hunk 2]
```diff
@@ -102,28 +101,16 @@
     <version.jakartaee-api>9.1-M2</version.jakartaee-api>
     <version.tomee-patch-plugin>0.9</version.tomee-patch-plugin>
 
-    <!--
-      JavaMail is both API and IMPL so we don't have them in the jakartaee-api uber jar.
-      We decided to add them here in the project so we can patch/update the 2 libraries without having
```
Comment (rzo1) [inline pom.xml:107]: Guess we should keep these comments

File: pom.xml [hunk 3]
```diff
@@ -102,28 +101,16 @@
...
-      to update and release the jakartaee-api.jar. Different lifecycle.
-    -->
-    <geronimo-mail_2.1_spec.version>1.0.0-M1</geronimo-mail_2.1_spec.version>
-    <geronimo-mail_2.1_provider.version>1.0.0</geronimo-mail_2.1_provider.version>
```
Comment (rzo1) [inline pom.xml:111]: Any reason for dropping one of these versions? Guess we need to check, what is needed / required.

File: pom.xml (hunk 1 above)
Comment (sultan) [reply pom.xml]: The name was incorrect, my mistake. now corrected

File: pom.xml (hunk 3 above)
Comment (sultan) [reply pom.xml:111]: relevant question, they are not dropped but moved, i redid the PR again and fixed the props names

File: pom.xml (hunk 2 above)
Comment (rzo1) [reply pom.xml:107]: Moving away from Geronimo API + Impls need to be discussed on the list. 

I wouldn't change that in this PR. We should keep it as ist and do the cleanup/renaming. If we have consensus on the list, we might switch away.

Comment (sultan) [reply pom.xml:107]: to be clear my intent is to keep Geronimo Impl. and swtich only the API.
but i get your point and will discuss that on the mailing list.

Comment (rzo1) [reply pom.xml:107]: There is also an ongoing discussion regarding Geronimo components on their list. There might be changes in the near future (attic, etc), but that discussion hasn't ended yet. So we might need to discuss / change things in the future.

**Review by rzo1 — APPROVED** [review]

**Pull request merged**